### ISSUE TITLE
[Annotation] Show case annotations after a user types a preposition

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-Android#275]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Android/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Android's [Code of Conduct](https://github.com/scribe-org/Scribe-Android/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue is the first to complete the case declension functionality within Scribe keyboards, which currently is a feature for German and Russian keyboards, but will be needed for others in the future. In these languages, after a preposition the speaker then needs to be using a new case such as accusative, dative, genitive and many others. An example of this in English is `With "whom" are you going to the library`? "With" puts us in a different case (no idea if this is defined for English, but for German it would be dative). In Scribe keyboards we thus show the case that would follow each preposition after the user types one.

- Note that this issue is blocked by #135 and #136 

The functionality for this feature is:

- The user types a preposition
- The user presses space
- The first autosuggestion key is replaced with an annotation in the style that will be developed in #135
    - The key is black for light mode and white for light mode with the opposite color for text
    - The key should be styled as below (rectangular with rounded corners and a shadow to make it clear it's a button)
- If there is more than one case that possibly follows a preposition, then all are shown on the annotation

An example for the German Scribe keyboard is (The German for "with", "mit", is dative case):

![Simulator Screenshot - iPhone 16 Pro - 2024-12-15 at 20 36 01](https://github.com/user-attachments/assets/3cbdfe58-cca7-4255-bc52-e86f0266c7eb)

### Contribution

Happy to assist with this and review once a PR is up! 🚀

**Comment by angrezichatterbox** [comment]
I will be working on this :)

**Comment by angrezichatterbox** [comment]
Just to confirm a few things 

For Dative we would display Dat 
For Accusative we would display Akk 
For Genitive we would display Gen 

Am I right ?

**Comment by andrewtavis** [comment]
Exactly, except for accusative it's ACC as a base :) There are also German and Russian versions of them, which can be found [here](https://github.com/scribe-org/Scribe-iOS/blob/main/Keyboards/KeyboardsBase/ScribeFunctionality/Annotate.swift#L30). So there's the base, and we should convert them over using the dictionary there 😊

Thanks for checking in, @angrezichatterbox!

**Comment by angrezichatterbox** [comment]
> Exactly, except for accusative it's ACC as a base :) There are also German and Russian versions of them, which can be found [here](https://github.com/scribe-org/Scribe-iOS/blob/main/Keyboards/KeyboardsBase/ScribeFunctionality/Annotate.swift#L30). So there's the base, and we should convert them over using the dictionary there 😊
> 
> Thanks for checking in, [@angrezichatterbox](https://github.com/angrezichatterbox)!

Okay Thanks. I think I should do the same for the Noun annotations as well.

**Comment by andrewtavis** [comment]
Yes exactly :) Would be good if the PR for this would include that as well 😊

**Comment by andrewtavis** [comment]
Closed by #306 🚀 Thanks for the great work here and the care you put into the styling, @angrezichatterbox!
